group/assignment5/AlphabetFromDictionary.cpp

```diff
@@ -1,120 +1,175 @@
 #include "AlphabetFromDictionary.hpp"
 
 AlphabetFromDictionary::AlphabetFromDictionary(const vector<string> &dictionary) : dictionary(dictionary) {
-    this->setDictionary(dictionary);
+	this->setDictionary(dictionary);
 }
 
 void AlphabetFromDictionary::setDictionary(const vector<string> &dictionary) {
-    this->dictionary = dictionary;
-    this->initializeClassVars();
-    this->findAlphabet();
+	this->dictionary = dictionary;
+	this->initializeClassVars();
+	this->findAlphabet();
 }
 
 vector<char> AlphabetFromDictionary::getAlphabet() {
-    return this->orderedAlphabet;
+	return this->orderedAlphabet;
 }
 
 bool AlphabetFromDictionary::isDictionaryConsistent() {
-    return !this->isGraphCyclic;
+	return !this->isGraphCyclic;
 }
 
 void AlphabetFromDictionary::initializeClassVars() {
-    this->findAlphabetCharacters();
-    this->isGraphCyclic = false;
-    this->orderedAlphabet.clear();
-    this->charPriorityGraph.clear();
-    for (auto c : this->alphabetCharacters) {
-        this->charPriorityGraph[c] = vector<bool>(NUM_OF_NATIVE_CHARS, false);
-    }
+	this->findAlphabetCharacters();
+	this->isGraphCyclic = false;
+	this->orderedAlphabet.clear();
+	this->charPriorityGraph.clear();
+	for (auto c : this->alphabetCharacters) {
+		this->charPriorityGraph[c] = vector<bool>(NUM_OF_NATIVE_CHARS, false);
+	}
+	this->inDegree.resize(NUM_OF_NATIVE_CHARS, 0);
 }
 
 void AlphabetFromDictionary::findAlphabetCharacters() {
-    this->alphabetCharacters.clear();
-    vector<bool> isCharInDictionary(NUM_OF_NATIVE_CHARS, false);
-    for (string &word : this->dictionary) {
-        for (char c : word) {
-            if (!isCharInDictionary[c]) {
-                isCharInDictionary[c] = true;
-                this->alphabetCharacters.push_back(c);
-            }
-        }
-    }
+	this->alphabetCharacters.clear();
+	vector<bool> isCharInDictionary(NUM_OF_NATIVE_CHARS, false);
+	for (auto word : this->dictionary) {
+		for (char c : word) {
+			if (!isCharInDictionary[c]) {
+				isCharInDictionary[c] = true;
+				this->alphabetCharacters.push_back(c);
+			}
+		}
+	}
 }
 
-int AlphabetFromDictionary::findFirstMismatchIndex(const string &str1, const string &str2) {
-    int index = 0;
-    while (index < str1.size() && index < str2.size() && str1[index] == str2[index])
-        index++;
-    return (index < str1.size() && index < str2.size()) ? index : -1;
+int AlphabetFromDictionary::findFirstMismatchIndex(string &str1, string &str2) {

```

Comment:
Is there a reason you remeved the "const" from the function declaration or was it (ctrl+k+d)?